mypy/newsemanal/semanal.py

```diff
@@ -3139,12 +3139,7 @@ def visit_with_stmt(self, s: WithStmt) -> None:
                         new_types.append(analyzed)
                         self.store_declared_types(n, analyzed)
 
-        # Reverse the logic above to correctly reassign target_type
-        if new_types:
-            if len(s.target) == 1:
-                s.target_type = new_types[0]
-            elif isinstance(s.target_type, TupleType):
-                s.target_type = s.target_type.copy_modified(items=new_types)
+        s.analyzed_types = new_types
```

Comment:
It looks like `analyzed_types` is only used in `deps.py`. Maybe we can update the logic there, so that we don't need this additional attribute? (Individual items can be already accessed on corresponding target `Var`s.)

Reply:
I considered this but the logic to support different kinds of lvalues would be more complicated. Also, this is kind of consistent with how assignment statements work.

Reply:
That is, the approach in the PR is consistent.